lib/puppet/provider/package/yum.rb

```diff
@@ -106,7 +106,12 @@ def self.parse_updates(str)
   end
 
   def self.update_to_hash(pkgname, pkgversion)
-    name, arch = pkgname.split('.')
+    # pkgname will contain dot (".") characters. The last dot character in the
+    # string will be the separator between the name of the package and the package
+    # architecture. Separate out those two values and assign them to variables.
+    splitindex = pkgname.rindex('.')
+    name = pkgname.slice(0..(splitindex - 1))
+    arch = pkgname.slice((splitindex + 1)..-1)
```

Comment:
This seems consistent with yum parsing of rpm filenames, split at right-most dot: https://github.com/rpm-software-management/yum/blob/043e869b08126c1b24e392f809c9f6871344c60d/rpmUtils/miscutils.py#L313. But perhaps we should check that `splitindex` isn't nil, otherwise we get:

```
irb(main):008:0> name.slice(0...nil)
ArgumentError: bad value for range
```